pkg/reconciler/revision/reconcile_resources.go

```diff
@@ -103,6 +103,19 @@ func (c *Reconciler) reconcileDeployment(ctx context.Context, rev *v1alpha1.Revi
 					break
 				}
 			}
+			// If RevisionConditionResourcesAvailable is false, with reason being "PodInitializing",
+			// check to see if the secrets are fetchable
+			if cond := rev.Status.GetCondition(v1alpha1.RevisionConditionResourcesAvailable); cond != nil && cond.Status == corev1.ConditionFalse && cond.Reason == "PodInitializing" {
```

Comment:
The reliance here on `PodInitializing` seems fragile. If we change our propagation logic or Kubernetes tries to make this error better this will unintentionally break and will likely be difficult to catch.

As an aside it makes me a bit sad we expose up `PodInitializing` through our Revision as a Reason since the user is not interacting with Pods.

Reply:
I agree. Do you have any suggestions on other approaches?